lib/app/ui.nit

```diff
@@ -67,6 +72,13 @@ redef class App
 	redef fun on_save_state do window.on_save_state
 end
 
+# Hook to create the first window shown to the user
+#
+# By default, a `Window` is created, which can be refined to customize it.
+# However, most apps should refine this method to return a different window,
+# this way the app can have more than one window.
```

Comment:
i dont understand le comment, there is more than one window, but how many root windows there is?

Reply:
There's a single root window, and one or more windows. So the idea is that for a quick small app you can work directly in `Window` but for larger apps you want to subclass it and redef `root_window` to select the desired root window.

I'm still not convinced that it is a good design but it does have its advantages.

I'll have to improve the doc, it will probably be at the module level or even at the package level to explain how the app is structured.